File: LV 1/JS/신규 아이디 추천.js
function solution(new_id) {
    var answer = new_id;
    // 1단계
    answer = new_id.toLowerCase()
    
    // 2단계
    let temp = ''
    for (let ch of answer) {
        if (('a' <= ch && ch <= 'z') ||
            ('0' <= ch && ch <= '9') ||
            ch === '-' || ch === '_' || ch === '.')  {
            temp = temp + ch
        }
    }
    answer = temp
    
    // 3단계
    let stack = []
    for (let ch of answer) {
        let l = stack.length;
        if (l !== 0) {
            if (ch === '.' && stack[l-1] === '.') {
                continue
            }
        } 
        stack.push(ch)
    }
    answer = stack.join('');
    
    // 4단계
    if (answer.length > 0 && answer[0] === '.')
        answer = answer.slice(1);
    if (answer.length > 0 && answer[answer.length-1] === '.')
        answer = answer.slice(0, -1);
    
    // 5단계
    if (answer.length === 0) {
        answer = 'a';
    }
    
    // 6단계
    if (answer.length >= 16) {
        answer = answer.slice(0, 15);
        if (answer[answer.length-1] === '.') 
            answer = answer.slice(0, -1);
    }
    
    if (answer.length <= 2) {
        answer = answer + answer[answer.length-1].repeat(3 - answer.length);
    }
    
    return answer;
}

File: LV 1/신규 아이디 추천.java
class Solution {
    public String solution(String new_id) {
        String answer = "";
        new_id = new_id.toLowerCase();
        String temp = "";
        for (int i=0; i<new_id.length(); i++) {
            char c = new_id.charAt(i);
            if (('a' <= c && c <= 'z') || 
               ('0' <= c && c <= '9') ||
               (c == '.') || (c == '-') || (c == '_')) {
                temp = temp + c;
            }
        }
        
        while (true) {
            String eliminated = temp.replace("..", ".");
            if (temp.equals(eliminated)) {
                temp = eliminated;
                break;
            }
            temp = eliminated;
        }
        
        if (temp.length() > 0 && temp.charAt(0) == '.') {
            temp = temp.substring(1, temp.length());
        }
        
        if (temp.length() > 0 && temp.charAt(temp.length()-1) == '.') {
            temp = temp.substring(0, temp.length()-1);
        }
        
        if (temp.length() == 0) {
            temp = "a";
        }
        
        if (temp.length() >= 16) {
            temp = temp.substring(0, 15);
            if (temp.charAt(temp.length()-1) == '.') {
                temp = temp.substring(0, 14);
            }
        }
        
        if (temp.length() <= 2) {
            char addCharacter = temp.charAt(temp.length() - 1);
            while (temp.length() != 3) {
                temp = temp + addCharacter;
            }
        }
        
        answer = temp;
        return answer;
    }
}
